### ISSUE TITLE
Can't open a new DM if the display name doesn't match the username

### ISSUE THREAD

**Issue by robbiemcmichael** [matterhorn-chat/matterhorn#415]
If you don't have an existing DM with a user and you try to open a new DM from the channel list (`C-g` mode) then this will fail if the display name doesn't match the username. The error produced is:

```
No such channel: "@Joe Bloggs"
```

The display name not matching the username might be a separate bug, but it recently started happening when our server had the [TeammateNameDisplay setting](https://docs.mattermost.com/administration/config-settings.html#teammate-name-display) set to `nickname_full_name`. This resulted in the nickname being shown if one existed, otherwise the username is shown.

Edit: Just found out the nickname/username display combination comes from #357 and is intentional, although it seems as though there's no way to turn it off if you don't run the server.

**Comment by jtdaugherty** [comment]
Thanks for the report! I investigated and found that maybe what you're running into here is that the `C-g` matching is case-sensitive. I think that's unnecessary and I'll change it so that it's case-insensitive as long as the input has no uppercase letters.

As for your edit: yes, the behavior is deliberate and is server-driven at the moment. I can look into using the user preference to take precedence. Does that take care of what you want?

**Comment by jtdaugherty** [comment]
> ... the C-g matching is case-sensitive. I think that's unnecessary and I'll change it so that it's case-insensitive as long as the input has no uppercase letters.

This is now done in 6b96b6718395d929f4b4a4d8070f28fbd1de425e.

**Comment by jtdaugherty** [comment]
I realized there might be another issue here, from

> This resulted in the nickname being shown if one existed, otherwise the username is shown.

Are you saying that you expect display name matches to be possible regardless of the display name mode that is active?

**Comment by robbiemcmichael** [comment]
This bug is a little harder to reproduce. It isn't a bug with the channel list matching, but instead with opening a new DM:

- If the client already has an existing DM open with the user you select, then it just opens that DM (regardless of whether a username or nickname is used): [src/State/Channels.hs#L845](https://github.com/matterhorn-chat/matterhorn/blob/6d95a73a7dde408fd17a15901590055002927747/src/State/Channels.hs#L845)
- If there isn't an existing DM, it will attempt to create a new channel for this. This works fine when the displayed name is just the regular username, but when the display name is a nickname it will attempt to create the channel using the display name which the server may reject: [src/State/Channels.hs#L839](https://github.com/matterhorn-chat/matterhorn/blob/6d95a73a7dde408fd17a15901590055002927747/src/State/Channels.hs#L839)

You could probably do other strange things like setting your nickname to someone else's display name, then I expect this bug would open a DM with the user whose username you spoofed instead.

**Comment by jtdaugherty** [comment]
Thanks for the code references; this is indeed a problem, as is the issue you raised about nickname/username clashes. I've looked into this and intend to fix it, but there are deeper problems that need to be resolved first. (Once those are resolved, we will get a fix for this for free.) I can't provide a timeline for that right now, though.

**Comment by jtdaugherty** [comment]
For posterity: this ticket depends on #350.

**Comment by jtdaugherty** [comment]
The issues raised in this ticket are now resolved in the `refactor/user-list` branch. Our next major release will include those changes. Thanks again!
